bc_obps/service/tests/test_user_operator_service.py

```diff
@@ -1,66 +1,241 @@
+        operator_instance: Operator = Operator(
+            business_structure=payload.business_structure,
+            cra_business_number=payload.cra_business_number,
+            bc_corporate_registry_number=payload.bc_corporate_registry_number,
+            status=Operator.Statuses.APPROVED,
+        )
+        UserOperatorService.save_operator(payload, operator_instance)
         assert len(Operator.objects.all()) == 1
-        assert Operator.objects.first().legal_name == "Example Legal Name"
-        assert len(ParentOperator.objects.all()) == 1
-        assert ParentOperator.objects.first().legal_name == "Example Parent Legal Name"
-        assert len(Address.objects.all()) == 4
+        assert Operator.objects.first().legal_name == payload.legal_name
+        assert Operator.objects.first().trade_name == payload.trade_name
+        assert Operator.objects.first().business_structure == payload.business_structure
+        assert Operator.objects.first().cra_business_number == payload.cra_business_number
+        assert Operator.objects.first().bc_corporate_registry_number == payload.bc_corporate_registry_number
+        assert Operator.objects.first().status == Operator.Statuses.APPROVED
+
+    @staticmethod
+    def test_list_user_operators_industry_users_are_not_authorized():
+        filters_1 = UserOperatorFilterSchema(
+            user_friendly_id="1",
+            status="pending",
+            user__first_name="john",
+            user__last_name="doe",
+            user__email="john.doe@test.com",
+            user__bceid_business_name="test business name",
+            operator__legal_name="test legal name",
+        )
+
+        # make sure only irc user can access this
+        industry_user = baker.make_recipe('registration.tests.utils.industry_operator_user')
+        with pytest.raises(Exception, match=UNAUTHORIZED_MESSAGE):
+            UserOperatorService.list_user_operators(
+                user_guid=industry_user.user_guid, filters=filters_1, sort_field="created_at", sort_order="asc"
+            )
+
+    @staticmethod
+    def test_list_user_operators():
+
+        # add some user operators
+        baker.make_recipe(
+            'registration.tests.utils.user_operator',
+            user=cycle(baker.make_recipe('registration.tests.utils.industry_operator_user', _quantity=5)),
+            role=UserOperator.Roles.ADMIN,
+            status=UserOperator.Statuses.APPROVED,
+            _quantity=5,
+        )
+        baker.make_recipe(
+            'registration.tests.utils.user_operator',
+            user=cycle(baker.make_recipe('registration.tests.utils.industry_operator_user', _quantity=5)),
+            role=UserOperator.Roles.ADMIN,
+            status=UserOperator.Statuses.DECLINED,
+            _quantity=5,
+        )
+        baker.make_recipe(
+            'registration.tests.utils.user_operator',
+            user=cycle(baker.make_recipe('registration.tests.utils.industry_operator_user', _quantity=5)),
```

Comment:
why is the `cycle` iterator needed here?

Reply:
To generate different users for each user_operator instance. 5 user for 5 user_operator.